app/services/announcements.ts

```diff
@@ -84,9 +86,29 @@ export class AnnouncementsService extends StatefulService<IAnnouncementsInfo> {
     return Date.now() - installationTimestamp < 1000 * 60 * 60 * 24 * 7;
   }
 
+  private get recentlyUpdatedTo017() {
+    const lastUpdatedVersion = this.patchNotesService.state.lastVersionSeen;
+
+    if (!lastUpdatedVersion) return false;
+
+    const minorVersionRegex = /^(\d+\.\d+)\.\d+$/;
+    const minorVersion = lastUpdatedVersion.match(minorVersionRegex);
+
+    if (!minorVersion || !minorVersion[1]) return false;
+    if (minorVersion[1] !== '0.17') return false;
+    if (!this.patchNotesService.state.updateTimestamp) return false;
+
+    const twoDaysAgo = Date.now() - 2 * 24 * 60 * 60 * 1000;
+
+    return Date.parse(this.patchNotesService.state.updateTimestamp) > twoDaysAgo;
```

Comment:
i feel like a lot of those early returns could be combined but non-blocking

Reply:
Yeah they could be.  Mentally I find functions like this easy to parse since each logical check is separated into a separate statement.  Big conditional statements are harder for me to follow and parse at a glance.  But certainly open to suggestions.